### ISSUE TITLE
Thefuck fixes itself but should go back in history

### ISSUE THREAD

**Issue by igorsantos07** [nvbn/thefuck#221]
It happened here a funny situation:

```
$ grep stuff .
grep: .: Is a directory
$ fuckl #typo
fuckl: command not found
$ fuck
fuck
fuck
fuck
[...]
```

I think fuck should understand when the fixed command is itself and go back one line in history, and fix that command instead? In that case, it would fix `fuckl` into `fuck`, and then fix `grep stuff .` into `grep -r stuff .`.

**Comment by scorphus** [comment]
Thanks, @igorsantos07. I can reproduce this one and I'm investigating it.

**Comment by scorphus** [comment]
Rereading this issue now I see it suggests more than fixing the loop. I'm going to think about it... but I think it would require more advanced shell aliases/functions.

**Comment by YarekTyshchenko** [comment]
Would be nice if it added the command it ran into the history itself. Possible to just append it to the history before it runs?

**Comment by scorphus** [comment]
Fixed in 9ba36c9
